src/solvers/flattening/boolbv_complex.cpp

```diff
@@ -47,6 +52,8 @@ bvt boolbvt::convert_complex(const exprt &expr)
 
 bvt boolbvt::convert_complex_real(const exprt &expr)
 {
+  PRECONDITION(expr.id() == ID_complex_real);
```

Comment:
Could such an expression be introduced, with a proper API? I have no idea what that is, though.

Reply:
It seems to be for getting the real part of a complex number. The introduction of `complex_real_exprt` and `complex_imag_exprt` has become a separate PR now: #2916.